src/programs/ectrans-benchmark.F90

```diff
@@ -397,12 +400,601 @@ program ectrans_benchmark
+endif
+
+ztstepavg = (ztstepavg/real(nproc,jprb))/real(iters,jprd)
+ztloop = ztloop/real(nproc,jprd)
+ztstep(:) = ztstep(:)/real(nproc,jprd)
+ztstepmed = get_median(ztstep)
+
+ztstepavg1 = (ztstepavg1/real(nproc,jprb))/real(iters,jprd)
+ztstep1(:) = ztstep1(:)/real(nproc,jprd)
+ztstepmed1 = get_median(ztstep1)
+
+ztstepavg2 = (ztstepavg2/real(nproc,jprb))/real(iters,jprd)
+ztstep2(:) = ztstep2(:)/real(nproc,jprd)
+ztstepmed2 = get_median(ztstep2)
+
+write(nout,'(a)') '======= Start of time step stats ======='
+write(nout,'(" ")')
+write(nout,'("Inverse transforms")')
+write(nout,'("------------------")')
+write(nout,'("avg  (s): ",f8.4)') ztstepavg1
+write(nout,'("min  (s): ",f8.4)') ztstepmin1
+write(nout,'("max  (s): ",f8.4)') ztstepmax1
+write(nout,'("med  (s): ",f8.4)') ztstepmed1
+write(nout,'(" ")')
+write(nout,'("Direct transforms")')
+write(nout,'("-----------------")')
+write(nout,'("avg  (s): ",f8.4)') ztstepavg2
+write(nout,'("min  (s): ",f8.4)') ztstepmin2
+write(nout,'("max  (s): ",f8.4)') ztstepmax2
+write(nout,'("med  (s): ",f8.4)') ztstepmed2
+write(nout,'(" ")')
+write(nout,'("Inverse-direct transforms")')
+write(nout,'("-------------------------")')
+write(nout,'("avg  (s): ",f8.4)') ztstepavg
+write(nout,'("min  (s): ",f8.4)') ztstepmin
+write(nout,'("max  (s): ",f8.4)') ztstepmax
+write(nout,'("med  (s): ",f8.4)') ztstepmed
+write(nout,'("loop (s): ",f8.4)') ztloop
+write(nout,'(" ")')
+write(nout,'(a)') '======= End of time step stats ======='
+write(nout,'(" ")')
+
+if (lstack) then
+  ! Gather stack usage statistics
+  istack = getstackusage()
+  if (myproc == 1) then
+    print 9001, istack
+    9001 format("Stack utilisation information",/,&
+         &"=============================",//,&
+         &"Task           size(bytes)",/,&
+         &"====           ===========",//,&
+         &"   1",11x,i10)
+
+    do i = 2, nproc
+      call mpl_recv(istack, ksource=nprcids(i), ktag=i, cdstring='ectrans_benchmark:')
+      print '(i4,11x,i10)', i, istack
+    enddo
+  else
+    call mpl_send(istack, kdest=nprcids(1), ktag=myproc, cdstring='ectrans_benchmark:')
+  endif
+endif
+
+
+!===================================================================================================
+! Cleanup
+!===================================================================================================
+
+deallocate(zgmv)
+deallocate(zgmvs)
+deallocate(sp3d)
+deallocate(zspsc2)
+deallocate(ivset)
+if (lprint_norms .or. ncheck > 0) then
+  deallocate(znormsp)
+  deallocate(znormsp1)
+  deallocate(znormvor)
+  deallocate(znormvor1)
+  deallocate(znormdiv)
+  deallocate(znormdiv1)
+  deallocate(znormt)
+  deallocate(znormt1)
+endif
+deallocate(ztstep)
+deallocate(ztstep1)
+deallocate(ztstep2)
+deallocate(nloen)
+
+!===================================================================================================
+
+call trans_release(nresol1)
+call trans_release(nresol2)
+call acc_present_dump
```

Comment:
`acc_present_dump` is probably a debugging leftover

Reply:
yes, ignore this file. I will completely revert it in the end, this is just to show how I tested it (I essentially copy pasted the whole file)

Reply:
reverted that commit, then we still have it in the history after merging. now it is passing the tests.